src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,845 @@
+                case HCI_OPCODE_LE_READ_PER_ADV_LIST_SIZE:
+                    // handle extended command
+                    if (hciCoreCb.extResetSeq) {
+                        /* send next extended command in sequence */
+                        (*hciCoreCb.extResetSeq)(pMsg, opcode);
+                    }
+                    break;
+
+                case HCI_OPCODE_LE_RAND:
+                    /* check if need to send second rand command */
+                    if (randCnt < (HCI_RESET_RAND_CNT - 1)) {
+                        randCnt++;
+                        HciLeRandCmd();
+                    } else {
+                        uint8_t addr[6] = { 0 };
+                        memcpy(addr, pMsg, sizeof(addr));
+                        DM_RAND_ADDR_SET(addr, DM_RAND_ADDR_STATIC);
+                        // note: will invoke set rand address
+                        set_random_static_address(addr);
+                    }
+                    break;
+
+                case HCI_OPCODE_LE_SET_RAND_ADDR:
+                    /* send next command in sequence */
+                    signal_reset_sequence_done();
+                    break;
+
+                default:
+#if defined(DEBUG)
+                    tr_info("Complete Event in reset seq with unknown opcode =0x%4X", opcode);
+#endif /* DEBUG */
+                    break;
+            }
+        } else if (*pMsg == HCI_CMD_STATUS_EVT) {
+            uint8_t status;
+            /* get status */
+            /* parse parameters */
+            pMsg += HCI_EVT_HDR_LEN;
+            status = *pMsg;
+            pMsg++;
+            pMsg++;                   /* skip num packets */
+            BSTREAM_TO_UINT16(opcode, pMsg);
+#if defined(DEBUG)
+            tr_info("Command Status event, status:%d, opcode=0x%4X", status, opcode);
+#endif /* DEBUG */
+        } else {
+            /**
+             * vendor specific event
+             */
+            if (pMsg[0] == VENDOR_SPECIFIC_EVENT) {
+                /* parse parameters */
+                pMsg += HCI_EVT_HDR_LEN;
+                BSTREAM_TO_UINT16(opcode, pMsg);
+#if defined(DEBUG)
+                tr_debug("Vendor specific event, opcode=0x%4X", opcode);
+#endif /* DEBUG */
+            } else {
+#if defined(DEBUG)
+                tr_info("Unknown event %d!", pMsg[0]);
+#endif /* DEBUG */
+            }
+        }
+#endif
+	    return 0;
+    }
+
+    uint16_t mbox_write(uint8_t type, uint16_t len, const uint8_t *pData)
+    {
+        TL_CmdPacket_t *bleCmdBuf = &BleCmdBuffer;
+
+/* this WA will clear the IPCC C1TOC2CSR 0x58000C0C else the next TX irq is not triggered 
+LL_C2_IPCC_ClearFlag_CHx( IPCC, HW_IPCC_SYSTEM_CMD_RSP_CHANNEL );*/
+
+
+        // Note: Until enum is avalable
+        // type 01 Command
+        // type 02 ACL DATA
+        // type 03 SCO Voice (not supported)
+        // type 04 event - uplink (not suported)
+#if defined(DEBUG)
+        tr_debug("mbox_write type:%d, len:%d", type, len);
```

Comment:
Can you implement / activate printf and add traces so that we can get a log to see a summary of the complete sequence that happens ?

Reply:
done with 'printf'